fix flaky test

`test_equality_returns_false_for_inequality()` fails for some examples produced by hypotesis if the quaternions-vectors are very close.

```
 ============================= test session starts ==============================
platform linux -- Python 3.9.6, pytest-6.2.4, py-1.10.0, pluggy-0.13.1
rootdir: /home/runner/work/pytf3d/pytf3d, configfile: pyproject.toml, testpaths: test
plugins: hypothesis-6.14.5, cov-2.12.1
collected 46 items

test/test_dummy.py .                                                     [  2%]
test/test_rotation.py .......F.....................................      [100%]

=================================== FAILURES ===================================
__________________ test_equality_returns_false_for_inequality __________________

    @given(q1=UnitQuaternionStrategy, q2=UnitQuaternionStrategy)
>   def test_equality_returns_false_for_inequality(q1: QUATERNION_T, q2: QUATERNION_T):

test/test_rotation.py:69: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

q1 = array([0., 0., 0., 1.])
q2 = array([0.0000000e+00, 0.0000000e+00, 1.0000889e-05, 1.0000000e+00])

    @given(q1=UnitQuaternionStrategy, q2=UnitQuaternionStrategy)
    def test_equality_returns_false_for_inequality(q1: QUATERNION_T, q2: QUATERNION_T):
        assume(not np.allclose(q1, q2, atol=1e-5))
        assume(not np.allclose(q1, -q2, atol=1e-5))
>       assert not Rotation(q1).almost_equal(Rotation(q2))
E       assert not True
E        +  where True = <bound method Rotation.almost_equal of Rotation(0.00000000 | 0.00000000, 0.00000000, 1.00000000)>(Rotation(0.00000000 | 0.00000000, 0.00001000, 1.00000000))
E        +    where <bound method Rotation.almost_equal of Rotation(0.00000000 | 0.00000000, 0.00000000, 1.00000000)> = Rotation(0.00000000 | 0.00000000, 0.00000000, 1.00000000).almost_equal
E        +      where Rotation(0.00000000 | 0.00000000, 0.00000000, 1.00000000) = Rotation(array([0., 0., 0., 1.]))
E        +    and   Rotation(0.00000000 | 0.00000000, 0.00001000, 1.00000000) = Rotation(array([0.0000000e+00, 0.0000000e+00, 1.0000889e-05, 1.0000000e+00]))

test/test_rotation.py:72: AssertionError
---------------------------------- Hypothesis ----------------------------------
Falsifying example: test_equality_returns_false_for_inequality(
    q1=array([0., 0., 0., 1.]),
    q2=array([0.0000000e+00, 0.0000000e+00, 1.0000889e-05, 1.0000000e+00]),
)

You can reproduce this example by temporarily adding @reproduce_failure('6.14.5', b'AXicY2DADaSuf5qNXUZwBkweAGDaBaY=') as a decorator on your test case
=========================== short test summary info ============================
FAILED test/test_rotation.py::test_equality_returns_false_for_inequality - as...
======================== 1 failed, 45 passed in 10.20s =========================
Error: Process completed with exit code 1.
```

--- a/src/pytf3d/_rotation.py
+++ b/src/pytf3d/_rotation.py
@@ -278,12 +278,15 @@ class Rotation:
     def almost_equal(self, other: "Rotation", eps: float = 1e-6) -> bool:
         """
         check if two Rotation objects represent the same rotation within tolerance
+
+        :param other: rotation to check against
+        :param eps: comparison tolerance, note that we compare the cosine tolerance and use eps**2 internally for that
         """
         # Check for cosine similarity between 4D-vectors (and keep in mind that q and -q are the same rotation).
         # This behaves numerically more stable than component-wise comparison.
         # See https://gamedev.stackexchange.com/a/75108 for more info.
         cosine_similarity = np.abs(np.dot(self._q, other.as_quaternion()))
-        return float(cosine_similarity) > 1.0 - eps
+        return float(cosine_similarity) > 1.0 - eps ** 2
 
     @staticmethod
     def _raise_if_not_expected_shape(a: np.ndarray, expected: Union[Tuple[int, ...], Set[Tuple[int, ...]]]) -> None:

--- a/test/test_rotation.py
+++ b/test/test_rotation.py
@@ -67,8 +67,8 @@ def test_equality_returns_true_for_equality(q: QUATERNION_T, diff: QUATERNION_T,
 
 @given(q1=UnitQuaternionStrategy, q2=UnitQuaternionStrategy)
 def test_equality_returns_false_for_inequality(q1: QUATERNION_T, q2: QUATERNION_T):
-    assume(not np.allclose(q1, q2, atol=1e-5))
-    assume(not np.allclose(q1, -q2, atol=1e-5))
+    assume(not np.allclose(q1, q2, atol=1e-6))
+    assume(not np.allclose(q1, -q2, atol=1e-6))
     assert not Rotation(q1).almost_equal(Rotation(q2))
 
 

